File: synthetic/go-gin-tasks/internal/repository/user_repo.go
package repository

import (
	"database/sql"
)

type UserRepo struct {
	db *sql.DB
}

func NewUserRepo(db *sql.DB) *UserRepo {
	return &UserRepo{db: db}
}

func (r *UserRepo) Create(email, password string) error {
	_, err := r.db.Exec("INSERT INTO users (email, password) VALUES (?, ?)", email, password)
	return err
}

func (r *UserRepo) FindByCredentials(email, password string) (int, error) {
	var id int
	err := r.db.QueryRow("SELECT id FROM users WHERE email = ? AND password = ?", email, password).Scan(&id)
	if err != nil {
		return 0, err
	}
	return id, nil
}

func (r *UserRepo) DeleteAll() error {
	_, err := r.db.Exec("DELETE FROM users")
	return err
}

File: synthetic/rust-axum-notes/src/repository/user_repo.rs
use sqlx::SqlitePool;

use crate::error::AppError;

pub struct UserRepo;

impl UserRepo {
    pub async fn find_by_credentials(pool: &SqlitePool, email: &str, password: &str) -> Result<Option<i64>, AppError> {
        let row: Option<(i64,)> = sqlx::query_as("SELECT id FROM users WHERE email = $1 AND password = $2")
            .bind(email)
            .bind(password)
            .fetch_optional(pool)
            .await?;
        Ok(row.map(|(id,)| id))
    }

    pub async fn delete_all(pool: &SqlitePool) -> Result<(), AppError> {
        sqlx::query("DELETE FROM users").execute(pool).await?;
        Ok(())
    }
}
